src/crimson/osd/main_config_bootstrap_helpers.h

```diff
@@ -0,0 +1,98 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <iostream>
+#include <fstream>
+#include <random>
+
+#include <seastar/core/future.hh>
+
+#include "common/ceph_argparse.h"
+#include "include/expected.hpp"
+
+namespace crimson::osd {
+
+void usage(const char* prog);
+
+inline uint64_t get_nonce()
+{
+  if (auto pid = getpid(); pid == 1 || std::getenv("CEPH_USE_RANDOM_NONCE")) {
```

Comment:
Rook was setting `2`. I guess this issue was tackled by setting `CEPH_USE_RANDOM_NONCE` but I'm not sure after the time.

Reply:
You mean getpid() returns 2 with rook?  The existing get_nonce() in main.cc seems to use precisely this logic, though.

Reply:
Will dig more after the long weekend. No regression, so no need to block.

Reply:
Gotcha, thanks.

Reply:
https://github.com/ceph/ceph/pull/42510/files

In crimson the problem got fixed by implementing the support for `CEPH_USE_RANDOM_NONCE` -- it wasn't there before.